sdk/batch/Microsoft.Azure.Batch.Conventions.Files/src/TaskOutputStorage.cs

```diff
@@ -51,56 +49,29 @@ public class TaskOutputStorage
         /// <remarks>The container must already exist; the TaskOutputStorage class does not create
         /// it for you.</remarks>
         public TaskOutputStorage(Uri jobOutputContainerUri, string taskId)
-            : this(CloudBlobContainerUtils.GetContainerReference(jobOutputContainerUri), taskId, null)
+            : this(CloudBlobContainerUtils.GetContainerReference(jobOutputContainerUri), taskId)
         {
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="JobOutputStorage"/> class from a storage account,
         /// job id, and task id.
         /// </summary>
-        /// <param name="storageAccount">The storage account linked to the Azure Batch account.</param>
+        /// <param name="blobClient">The blob client linked to the Azure Batch storage account.</param>
         /// <param name="jobId">The id of the Azure Batch job containing the task.</param>
         /// <param name="taskId">The id of the Azure Batch task.</param>
         /// <remarks>The job output container must already exist; the TaskOutputStorage class does not create
         /// it for you.</remarks>
-        public TaskOutputStorage(CloudStorageAccount storageAccount, string jobId, string taskId)
-            : this(CloudBlobContainerUtils.GetContainerReference(storageAccount, jobId), taskId, null)
+        public TaskOutputStorage(BlobServiceClient blobClient, string jobId, string taskId)
+            : this(CloudBlobContainerUtils.GetContainerReference(blobClient, jobId), taskId)
         {
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="JobOutputStorage"/> class from a task id and
-        /// a URL representing the job output container.
-        /// </summary>
-        /// <param name="jobOutputContainerUri">The URL in Azure storage of the blob container to
-        /// use for outputs associated with this job. This URL must contain a SAS (Shared Access
-        /// Signature) granting access to the container, or the container must be public.</param>
-        /// <param name="taskId">The id of the Azure Batch task.</param>
-        /// <param name="storageRetryPolicy">The retry policy for storage requests.</param>
-        /// <remarks>The container must already exist; the TaskOutputStorage class does not create
-        /// it for you.</remarks>
-        public TaskOutputStorage(Uri jobOutputContainerUri, string taskId, IRetryPolicy storageRetryPolicy)
-            : this(CloudBlobContainerUtils.GetContainerReference(jobOutputContainerUri), taskId, storageRetryPolicy)
```

Comment:
Current Storage SDK does not provide interoperability with Retry Policies. You'll need to set RetryOptions on the BlobServiceClient. This will be documented in the ChangeLog